dtls.c

```diff
@@ -2005,11 +2005,9 @@ dtls_asn1_len(uint8 **data, size_t *data_len)
     (*data_len)--;
     if (octets && *data_len == 0)
```

Comment:
If the issue is an invalid encoded ASN.1 length, (in the example it indicates a length of 7 bytes, but only 2 are available), an alternative solution maybe to check this ahead:

```
if (octets > *data_len)
```

Reply:
Yes, an upfront check is even better. I will update the code in a minute.

Reply:
And please, against "main" :-).

Reply:
New code pushed as fixup for later squash.